cdap-docs/developers-manual/source/data-exploration/filesets.rst

```diff
@@ -0,0 +1,86 @@
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", schema.toString())
+      .build());
+
+These Dataset properties map directly to table properties in Hive. 
+In fact, the create table statement that will be generated for the table above is::
+
+  CREATE EXTERNAL TABLE cdap_user_myfiles(
+    user string,
+    item_id int,
+    price double)
+  ROW FORMAT SERDE "org.apache.hadoop.hive.serde2.avro.AvroSerDe"
+  STORED AS INPUTFORMAT "org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat"
+  OUTPUTFORMAT "org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat"
+  LOCATION "/cdap/mylocation"
+  TBLPROPERTIES (
+    "avro.schema.literal"="{\"type\": \"record\", \"name\": \"stringBody\", \"fields\": [{ \"name\":\"ts\", \"type\":\"long\" }, { \"name\":\"body\", \"type\":\"string\" } ] }"
+  );
+
+Please see the `Hive Language Manual
+<https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Create/Drop/TruncateTable>`__
+for more information.
+
+Limitations
+-----------
+As exploration is tied directly to Hive, file exploration has several limitations that mirror limitations in Hive.
+
+- All explorable files must be of the same format.
+- All explorable files must be in a format known to the version of Hive you are using.
+- Some versions of Hive may try to create a temporary staging directory at the table location when executing queries.
+  If you are seeing permissions errors, try setting ``hive.exec.stagingdir`` in your Hive configuration to ``/tmp/hive-staging``.
+
+A ``FileSet`` has some additional limitations that the ``TimePartitionedFileSet`` does not have.
+
+- Hive tables created are not partitioned. This means any query will perform a full table scan.
+- Only files at the base location of the ``FileSet`` are visible to queries. Directories are not read.
+  Since Mapreduce writes output files to a directory, if you want Mapreduce output to be explorable you
+  must move all output files to the base location. 
+
+If you wish to use Impala to explore a ``FileSet`` or ``TimePartitionedFileSet``, there are several
+additional restrictions you must keep in mind.
+
+- Impala only supports scalar types. See `Data Types <http://www.cloudera.com/content/cloudera/en/documentation/cloudera-impala/latest/topics/impala_avro.html#avro_data_types_unique_1>`__ for more detail.
+- If your underlying data contains non-scalars, you cannot tell Impala to use a different read schema of just scalars.
+  For example, if you have Avro files that contain a map field, you cannot simply leave out the field when specifying the table schema.
+- Impala caches table metadata, which clients must invalidate when there are changes. 
+  You have to issue the INVALIDATE METADATA [tablename] command, whenever table metadata changes.
```

Comment:
You have to issue an ``INVALIDATE METADATA [tablename]`` command whenever the table metadata changes.
You'll need to invalidate metadata if a new table or table partition is added. Otherwise, Impala will use the 
cached table metadata, preventing you from seeing the changes. If data is added to a table without changing the metadata (such as when adding a partition), then you need to issue 
a ``REFRESH [tablename]`` command to force Impala to see the changes. 
Though Impala also caches info on table files and blocks, any calls to the ``REFRESH`` command will cause it to re-read the information.